web/js/modules/geosearch/reducers.js

```diff
@@ -0,0 +1,51 @@
+import { assign as lodashAssign } from 'lodash';
+import {
+  CLEAR_COORDINATES,
+  SELECT_COORDINATES_TO_FLY,
+  TOGGLE_REVERSE_GEOCODE_ACTIVE,
+  TOGGLE_SHOW_GEOSEARCH,
+  UPDATE_ACTIVE_MARKER,
+} from './constants';
+import { getLocalStorageCollapseState } from './util';
+
+const localStorageCollapseState = getLocalStorageCollapseState();
+export const geosearchState = {
+  isCoordinateSearchActive: false,
+  isExpanded: !localStorageCollapseState,
+  coordinates: [],
+  activeMarker: null,
+  reverseGeocodeResults: null,
+};
+
+export function geosearchReducer(state = geosearchState, action) {
+  switch (action.type) {
+    case UPDATE_ACTIVE_MARKER:
+      return lodashAssign({}, state, {
+        activeMarker: action.value,
+        reverseGeocodeResults: action.reverseGeocodeResults,
+      });
```

Comment:
We shouldn't need to use `lodashAssign` for this, the spread operator in ES6 will handle this natively:

```
return { 
    ...state,
    activeMarker: action.value,
    reverseGeocodeResults: action.reverseGeocodeResults,
};
```